### PR TITLE
Added Windows build for detectron2

### PR DESCRIPTION
**Pull request by sarthakforwet**
PR from the outcome of the issue #1712 .

Updated:
- Added `<time.h>` header file in `cocoeval.cpp`.
- Changed `localtime_r` function to `localtime_s`.
- Updated config.yml file to add job for windows (cpu) build of detectron2.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -46,6 +68,16 @@ install_dep: &install_dep
         pip install --progress-bar off opencv-python pytest-xdist
         pip install --progress-bar off torch torchvision
 
+install_dep_win: &install_dep_win
+  - run:
+      name: Install Dependencies on Windows
+      command: |
+        pip install cython
```
Comment (ppwwyyxx) [inline .circleci/config.yml:75]: From the current failures in https://circleci.com/gh/facebookresearch/detectron2/1690?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link it looks like `numpy` has to be installed here as well. `fetch_build_eggs` for some reason does not work on windows

**Comment by sarthakforwet** [comment]
image

image

This is the error I am getting for `FasterRCNNTest.test_flop` and `RetinaNetTest.test_flop`. 
@ppwwyyxx can you please tell me how can this error be solved? Why value of `res` is becoming zero?

**Comment by ppwwyyxx** [comment]
If test failure is difficult to resolve now, it's fine to first add the CI with builds only but not unittests. At least it will make sure the code compiles on windows (and functional for the most parts)

**Comment by sarthakforwet** [comment]
Due to my college semester, I was unable to lookup on the unittests issue but I think you are right about merging without unittests. However I would like to create an issue (on your call) about the failure of test cases. I would also keep working on that issue.

**Comment by sarthakforwet** [comment]
Further Work: 

- [ ] Add Windows build for GPU
- [ ] Add Unittests to Windows Builds

**Comment by ppwwyyxx** [comment]
After taking a closer look it turns out that the build actually did not succeed in CI. In "installing dependencies" it failed for pycocotools with the message:
> Installing collected packages: cycler, pillow, kiwisolver, python-dateutil, certifi, pyparsing, matplotlib, pycocotools
  Found existing installation: certifi 2019.3.9
Cannot uninstall 'certifi'. It is a distutils installed project and thus we cannot accurately determine which files belong to it which would lead to only a partial uninstall.

As a result it causes this in the end:
> detectron2._C          failed to import

which says detectron2 is not compiled

**Comment by sarthakforwet** [comment]
I think its successfully built now @ppwwyyxx. I've added a force installation of the system to newest version of `certifi`.

**Comment by ppwwyyxx** [comment]
nice fix! thanks!

**Comment by sarthakforwet** [comment]
@ppwwyyxx Congratulations on 2.0.2 version of pycocotools. 💥💥

**Pull request closed without merging**